partcad/tests/unit/test_ocp_serialize_compression.py

```
@@ -0,0 +1,133 @@
+    for _ in range(3):
+        ocp_serialize.encode_shape(_box())
+
+    assert capsys.readouterr().err.count("zstd is not available") == 1
+
+
+def test_decoding_a_compressed_payload_without_zstd_says_what_is_missing(monkeypatch):
+    """The one case that cannot degrade: a compressed payload and no zstd."""
+    encoded = ocp_serialize.encode_shape(_box())
+    monkeypatch.setattr(ocp_serialize, "_zstd_decompress", None)
+
+    with pytest.raises(RuntimeError, match="backports.zstd"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)